### ISSUE TITLE
HUD is rendered underneath vignette, making it very dark

### ISSUE THREAD

**Issue by GammaBoost** [SoRadGaming/Simple-HUD-Enhanced#69]
**Describe the bug**
If you have the vignette enabled, any HUD elements that are in the corners will be significantly dimmed when the vignette is darker.

**To Reproduce**
Steps to reproduce the behaviour:
1. Make sure vignette is not disabled in Sodium, or modified in your resource pack
2. Have HUD elements in the corners
3. Put your character in a dark in-game environment where the vignette is darker than usual 
4. See that the HUD elements are dimmed

**Expected behaviour**
HUD should be rendered on top of vignette, as to not be affected by vignette

**Minecraft (please complete the following information):**
 - Game Version: 1.21.3
 - Mod Version: 4.7.2

**Additional context**
I have installed the Sodium mod

**Comment by SoRadGaming** [comment]
interesting, I'll keep the ticket open, in case anyone else has the issue, so we can find a solution.

**Comment by SoRadGaming** [comment]
Yeah, the z index of my mod is defined to be the same as Hud, other mods like minimap override this, I'm not sure If I like the behaviour 
![2024-12-10_02 00 14](https://github.com/user-attachments/assets/c4df3d8d-f179-4a2c-85c0-9c90a2f644a0)

I could fix this, but I'm not sure if I should, I kinda like that onscreen is effecting the hud

**Comment by SoRadGaming** [comment]
After more thought, ill render the text over everything, so it will be coming in a future update to fix this issue, hopefully v4.7.5

**Comment by PrinceLikesCookies** [comment]
Yeah the text is very dark even though the color is set as a bright color 

![Image](https://github.com/user-attachments/assets/ca90753e-6400-4344-965e-b7e44fbf44e4)

**Comment by GammaBoost** [comment]
I figured out my issue. It was the Vanilla Tweaks texture pack, specifically the "Dark UI" and "Transparent UI" options. Disabling those fixed it.

**Comment by BeeFox-sys** [comment]
I am still experiencing this bug, i do have vanilla tweaks dark ui however turning it off does not fix the issue

**Comment by SoRadGaming** [comment]
Yeah, I also have this problem. Before addressing this, I wanted to rewrite the render system, as it would potentially break the fix anyway, ill look into fixing this for the next update

**Comment by GammaBoost** [comment]
I think I figured out how Vanilla Tweaks causes this issue. It uses a VSH (vertex shader) file to replace the black text in the inventory to white. It uses the text's Z coordinates to choose the right layer. I don't know why that affects the in-game HUD but deleting the file does fix the issue. That's not really useable though so I just disable Dark UI.

If you still have the problem, disable all texture packs and then try each one individually to figure out which one causes the conflict. 

Otherwise you can just temporarily disable the vignette entirely through sodium for now.
